src/Numeric/Half.hs

```diff
@@ -190,3 +191,6 @@ instance Num Half where
   abs = toHalf . abs . fromHalf
   signum = toHalf . signum . fromHalf
   fromInteger a = toHalf (fromInteger a)
+
+instance Lift Half where
+  lift = return . LitE . FloatPrimL . toRational 
```

Comment:
We could put the Half constructor in then apply it to a word literal underneath.